src/main/java/com/sphy/hotelmanagementapplication/service/RoomService.java

```diff
@@ -85,45 +94,45 @@ public List<RoomDTO> saveRooms(List<RoomDTO> roomsDTO) throws ApiRequestExceptio
         roomRepository.saveAll(rooms);
 
         List<RoomDTO> roomDTOS = new ArrayList<>();
-        for (Room room:rooms){
-            roomDTOS.add(roomToRoomDTO.converter(roomRepository.findById(room.getId()).get()));
-        }
+
+        rooms.forEach(room -> roomsDTO.add(roomToRoomDTO.converter(room)));
+
         return roomDTOS;
     }
 
     /***
      * counts all the rooms in the database
      * @return the number of rooms that exists in the database
      */
-    public int countRooms(){
+    public int countRooms(Long id){
 
-        return roomRepository.countAll();
+        return roomRepository.countAll(id);
     }
 
     /***
      * get all rooms
      * @return a list of all rooms
      * @throws ApiRequestException if no room is saved
      */
-    public List<RoomDTO> getRooms(Integer pageNo, Integer pageSize, String sortBy) throws ApiRequestException {
+    public List<RoomDTO> getRooms(Integer pageNo, Integer pageSize, String sortBy, Long id) throws ApiRequestException {
 
         Pageable paging = PageRequest.of(pageNo, pageSize, Sort.by(sortBy));
 
-        Page<Room> pageResult = roomRepository.findAll(paging);
+        Page<Room> pageResult = roomRepository.findAllRoomsByOwner(id, paging);
 
         List<RoomDTO> roomDTOS = new ArrayList<>();
 
-        for (Room room: pageResult.getContent()){
-            roomDTOS.add(roomToRoomDTO.converter(room));
-        }
+        pageResult.getContent().forEach(room -> roomDTOS.add(roomToRoomDTO.converter(room)));
```

Comment:
I have the notion that this might produce NullPointerException, please add a check if getContent() is not null